orcidhub-core/authcontroller.py

```diff
@@ -393,3 +437,265 @@ def reset_db():
+        return redirect(url_for("viewmembers"))
+
+    client = OAuth2Session(user.organisation.orcid_client_id, token={"access_token": orcidToken.access_token})
+
+    headers = {'Accept': 'application/vnd.orcid+json'}
+    resp = client.get('https://api.sandbox.orcid.org/v2.0/' + user.orcid + '/employments', headers=headers)
+
+    data = json.loads(resp.text)
+    return render_template("viewEmpDetails.html", empdetails=data['employment-summary'],
+                           userEmail=userEmail, organisationId=organisationId)
+
+
+@app.route("/editempdetails", methods=['GET', 'POST'])
+@roles_required(Role.ADMIN)
+def edit_emp_details():
+    userEmail = request.args.get("email")
+    organisationId = request.args.get("organisationId")
+    putcode = request.args.get("putcode")
+    user = User.get(email=userEmail, organisation_id=int(organisationId))
+    dataToUpdate = ""
+    form = EmploymentDetailsForm()
+    if request.method == 'GET':
+        orcidToken = None
+        try:
+            orcidToken = OrcidToken.get(user=user, org=user.organisation, scope=scope_read_limited)
+        except:
+            flash("The member has not given read and update permissions", "warning")
+            return redirect(url_for("link"))
+
+        client = OAuth2Session(user.organisation.orcid_client_id, token={"access_token": orcidToken.access_token})
+
+        headers = {'Accept': 'application/vnd.orcid+json'}
+        resp = client.get('https://api.sandbox.orcid.org/v2.0/' + user.orcid + '/employments', headers=headers)
+        data = json.loads(resp.text)['employment-summary']
+
+        dataToUpdate = ""
+        for empdetail in data:
+            if empdetail['put-code'] == int(putcode):
+                dataToUpdate = empdetail
+                break
+
+        start_date = datetime(int(dataToUpdate['start-date']['year']['value']),
+                              int(dataToUpdate['start-date']['month']['value']),
+                              int(dataToUpdate['start-date']['day']['value']))
+
+        end_date = datetime(int(dataToUpdate['end-date']['year']['value']),
+                            int(dataToUpdate['end-date']['month']['value']),
+                            int(dataToUpdate['end-date']['day']['value']))
+
+        department = dataToUpdate['department-name']
+        role = dataToUpdate['role-title']
+        institution = dataToUpdate['organization']['name']
+        city = dataToUpdate['organization']['address']['city']
+        regionName = dataToUpdate['organization']['address']['region']
+        country = dataToUpdate['organization']['address']['country']
+        form = EmploymentDetailsForm(state=regionName, country=country, city=city, institution=institution,
+                                     role=role, start_date=start_date, end_date=end_date, department=department)
+        return render_template("addEmploymentDetails.html", form=form)
+    elif request.method == 'POST':
```

Comment:
the same, all viewing, editing, and adding a new entry can be be and should done for the sake of "dry"-ness done in a single view. The whole purpose of WTForms is to provide the binding to "model"/original data and request.POST...

See: 

https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/blob/emp/orcidhub-core/views.py#L46
https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/blob/emp/orcidhub-core/views.py#L79
https://github.com/Royal-Society-of-New-Zealand/NZ-ORCID-Hub/blob/emp/orcidhub-core/views.py#L89